src/story_protocol_python_sdk/resources/IPAccount.py

```diff
@@ -153,4 +156,85 @@ def owner(self, ip_id: str) -> str:
         except ValueError:  # Catch ValueError from to_checksum_address
             raise ValueError(f"Invalid IP id address: {ip_id}")
         except Exception as e:
-            raise e
\ No newline at end of file
+            raise e
+
+    def setIpMetadata(self, ip_id: str, metadata_uri: str, metadata_hash: str, tx_options: dict = None) -> dict:
+        """Sets the metadataURI for an IP asset.
+
+        :param ip_id str: The IP ID to set metadata for.
+        :param metadata_uri str: The metadata URI to set.
+        :param metadata_hash str: The metadata hash.
+        :param tx_options dict: [Optional] The transaction options.
+        :returns dict: A dictionary with the transaction hash.
+        :raises ValueError: If the IP ID is invalid or not registered.
+        """
+        try:
+            if not self._is_registered(ip_id):
```

Comment:
for future reference, i think we can actually skip this check (its something we are looking to do in tsdk as well) if the poc call later handles this error already.